tests/window_management_tests/test_minimal_window_manager.cpp

```diff
@@ -582,14 +583,177 @@ TEST_F(MinimalWindowManagerTest,  when_a_window_is_focused_then_it_appears_above
     auto window2 = create_window(app2, spec);
     auto window3 = create_window(app3, spec);
 
-    request_focus(window1);
+    tools().select_active_window(window1);
 
     EXPECT_TRUE(focused(window1));
 
     EXPECT_TRUE(is_above(window1, window3));
     EXPECT_TRUE(is_above(window3, window2));
 }
 
+TEST_F(MinimalWindowManagerTest, if_active_window_is_removed_then_last_focused_window_is_focused)
+{
+    // Setup: Create three apps with a window for each app
+    auto const app1 = open_application("app1");
+    auto const app2 = open_application("app2");
+    auto const app3 = open_application("app3");
+    miral::WindowSpecification spec;
+    spec.size() = { geom::Width {100}, geom::Height{100} };
+    spec.depth_layer() = mir_depth_layer_application;
+    auto const window1 = create_window(app1, spec);
+    auto const window2 = create_window(app2, spec);
+    auto const window3 = create_window(app3, spec);
+    EXPECT_TRUE(focused(window3));
+
+    // Act: Close the focused window
+    tools().ask_client_to_close(window3);
+
+    // Expect: window2 to have focus
+    EXPECT_TRUE(focused(window2));
+    EXPECT_TRUE(is_above(window2, window1));
```

Comment:
[non blocking] I wonder how hard it would be to support the "fluent" style of assertions:

```suggestion
    EXPECT_THAT(window2, IsAbove(window1));
```

Reply:
I can set out to do transfer the window management stuff over on a followup pr! This seems like a reasonable direction though.